### ISSUE TITLE
SC2050 "expression is constant" when checking against a file

### ISSUE THREAD

**Issue by AlanDeSmet** [koalaman/shellcheck#2933]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2050
- My shellcheck version (`shellcheck --version` or "online"): online
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#! /bin/sh

if [ '!' -e outputfile ]; then 
        date > outputfile
fi
```

#### Here's what shellcheck currently says:

```
Line 3:
if [ '!' -e outputfile ]; then
         ^-- SC2050 (warning): This expression is constant. Did you forget the $ on a variable?
```

(It also reports a spurious SC2057, but that's discussed in #1984)

#### Here's what I wanted or expected to see:

Saying the expression is constant seems incorrect; the test is against the filesystem which is not typically constant. Potentially nothing could be said, and any file test could be considered non-constant. Alternatively, a more nuanced warning could be issued when there is a bare filename not in single quotes (single quotes would clarify that a non-variable was intended). For example:

```
SC#### (warning): This test is comparing to a an exact filename where a variable might have been intended.

If you meant the exact filename, consider:
    -e 'outputfile'
or if a variable was intended, perhaps:
    -e $outputfile
```
